drivers/regulator/regulator_cp9314.c

```diff
@@ -283,14 +269,75 @@ static int regulator_cp9314_get_error_flags(const struct device *dev,
 	return 0;
 }
 
+static int regulator_cp9314_write_lock(const struct device *dev,
+				       const enum cp9314_backdoor_keys key)
+{
+	const struct regulator_cp9314_config *config = dev->config;
+	struct regulator_cp9314_data *data = dev->data;
+	int ret;
+
+	if (data->allow_hw_i2c_lock == 0U) {
+		ret = i2c_reg_update_byte_dt(&config->i2c, CP9314_REG_LION_CFG_3,
+					     CP9314_ALLOW_HW_I2C_LOCK, CP9314_ALLOW_HW_I2C_LOCK);
+		if (ret < 0) {
+			return ret;
+		}
+
+		data->allow_hw_i2c_lock = true;
+	}
+
+	if ((uint8_t)key == data->backdoor_key) {
+		return 0;
+	} else {
+		return i2c_reg_write_byte_dt(&config->i2c, CP9314_REG_BACKDOOR_CTRL, (uint8_t)key);
+	}
+}
+
+static int regulator_cp9314_write_lock_init(const struct device *dev)
+{
+	const struct regulator_cp9314_config *config = dev->config;
+	struct regulator_cp9314_data *data = dev->data;
+	uint8_t value;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, CP9314_REG_LION_CFG_3, &value);
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->allow_hw_i2c_lock = FIELD_GET(CP9314_ALLOW_HW_I2C_LOCK, value);
```

Comment:
Is this bit set based on whether the PGPIO2 resistor is there or not?

Reply:
@aasinclair the write lock mechanism only active if both PGIO2 is biased appropriately and the ALLOW_HW_I2C bit is set.

Reply:
I'm a little confused about this being read here.
Is it expected to be set only by this SW driver?
Could you just set this at init based on hw_i2c_lock?

Reply:
Depending on the OTP variant used, this `ALLOW_HW_I2C` bit may or may not already be set coming out of reset or power on.

I read `ALLOW_HW_I2C` to figure out its state and if it is necessary to disable it during the rest of `regulator_cp9314_init`. Once `regulator_cp9314_init` is complete `ALLOW_HW_I2C` will be set by the first `regulator_cp9314_write_lock` call.

Reply:
Is the disabling of protection during init needed to speed up the initialisation process?

Reply:
Disabling the protection is not explicitly needed, but you are correct the intention is to speed-up/simplify init since that is several consecutive write operations occur.